### DISCUSSION TITLE
Etcd key value store retention planning

### DISCUSSION THREAD

**Discussion by allahshukur-ahmadzada** [grafana/mimir#4599] [Help and support]
We are planning to use etcd as key value store based on following [doc](https://grafana.com/docs/mimir/latest/operators-guide/configure/configure-hash-rings/#etcd) section. But it is not giving information about how we should calculate etcd retention/compaction time because while we test it after some time etcd grows and its restart takes long time. Anyone can help how we should calculate etcd compaction timing?

**Comment by pracucci** [comment]
I don't have much experience with etcd as backend for the ring, but at Grafana Labs we use etcd as backend for the HA deduplication and here is an example how we deploy it with the etcd operator:

https://github.com/grafana/mimir/blob/71c6b6807bcb111d68cb1c96f976334bd0d14d7e/operations/mimir-tests/test-defaults-generated.yaml#L1141-L1172

I think you may be particularly interested to this:

```
    - name: ETCD_AUTO_COMPACTION_RETENTION
      value: 1h
```
